File: frontend/src/types/game.ts
export type GameStatus = 'scheduled' | 'in_progress' | 'final'

export interface Team {
  id: string
  abbr: string
  name: string
  conference: string | null
  division: string | null
  logo_url: string | null
  primary_color: string | null
  secondary_color: string | null
}

export interface Game {
  id: string
  season: number
  week: number
  home_team: Team
  away_team: Team
  status: GameStatus
  nflfastr_game_id: string | null
  scheduled_at: string | null
  started_at: string | null
  final_home_score: number | null
  final_away_score: number | null
  venue: string | null
  home_wp: number | null
  away_wp: number | null
}

export interface GameDetail extends Game {
  play_count: number
}

File: backend/app/schemas/game.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.db.models.game import GameStatus
from app.schemas.team import TeamRead


class GameRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    season: int
    week: int
    home_team: TeamRead
    away_team: TeamRead
    status: GameStatus
    nflfastr_game_id: str | None = None
    scheduled_at: datetime | None
    started_at: datetime | None
    final_home_score: int | None
    final_away_score: int | None
    venue: str | None
    # Live WP is injected by the service layer, not stored on the Game model
    home_wp: float | None = None
    away_wp: float | None = None


class GameDetail(GameRead):
    """Extended game info returned by GET /games/{id}."""
    play_count: int = 0
